akka-actor-typed/src/main/scala/akka/actor/typed/Behavior.scala

```diff
@@ -44,6 +44,16 @@ sealed abstract class Behavior[T] { behavior ⇒
    */
   final def narrow[U <: T]: Behavior[U] = this.asInstanceOf[Behavior[U]]
 
+  /**
+   * Unsafe utility method for changing the type accepted by this Behavior;
+   * provided as an alternative to the universally available `asInstanceOf`, which
+   * casts the entire type rather than just the type parameter.
+   * Typically used to upcast a type, for instance from `Nothing` to some type `U`.
+   * Use it with caution, it may lead to a [[ClassCastException]] when you send a message
+   * to the resulting [[Behavior[U]]].
+   */
+  final def unsafeCast[U]: Behavior[U] = this.asInstanceOf[Behavior[U]]
```

Comment:
great！

Reply:
I introduced this once in https://github.com/akka/akka/pull/26024

Reply:
Should it be `unsafeUpcast` in the same way as `ActorRef.unsafeUpcast`. `narrow` should be used for the other direction. Are we sure this should be exposed to users, or should we start out with making it internal API?

Reply:
Not every cast is to a super-type, which is why it isn't `unsafeUpcast`.

I'm happy with making it internal API, which formulation do you prefer?  `@InternalApi`? `private[akka]`? Separate implicit class? Separate function?

Reply:
ok

```
  /**
   * INTERNAL API
   */
  @InternalApi private[akka]
```